### PR TITLE
MOTECH-2705: Added Integration Tests for post action parameter test module.

### PR DESCRIPTION
**Pull request by pkornowski**

### CODE REVIEW

**Comment by pkornowski** [comment]
test this please

File: modules/tasks/tasks/pom.xml [hunk 1]
```diff
@@ -61,6 +61,12 @@
             <artifactId>org.motechproject.gemini-blueprint-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.motechproject</groupId>
```
Comment (kmadej) [inline modules/tasks/tasks/pom.xml]: group id

File: modules/tasks/tasks/pom.xml [hunk 2]
```diff
@@ -61,6 +61,12 @@
...
+            <artifactId>motech-tasks-test-bundle</artifactId>
+            <version>0.29-SNAPSHOT</version>
```
Comment (kmadej) [inline modules/tasks/tasks/pom.xml]: use $motech.version

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class ActionParametersIT extends BasePaxIT {
```
Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java]: delete created task

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        Channel channel = loadChannel();
+        setUpSecurityContext("motech", "motech", "manageTasks");
+
+        channelService.addOrUpdate(channel);
+        waitForChannel(MDS_CHANNEL_NAME);
+    }
+
+    @Test
+    public void testActionWithPostActionParameters() throws InterruptedException, IOException, ActionNotFoundException {
+        long taskID = createTestTask();
```
Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java]: it will be better to use second action and check if post action params are correctly stored

Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersIT.java]: do not check PostActionParameterObject, it is implementation detail, you should check if TaskTestObject has proper field value after action executed

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersBundleIT.java [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+        taskID = createTestTask();
+
+        activateTrigger();
+        assertTrue(waitForTaskExecution(taskID));
+
+        List<TaskTestObject> fetchedTaskTestObjects = tasksTestService.getTaskTestObjects();
+
+        List<TaskTestObject> expectedTaskTestObject = prepareExpectedTaskTestObject();
+
+        deleteTask(taskID);
```
Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersBundleIT.java]: deleteTask and removeChannels should be done in method invoked after every test

File: modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java [hunk 6]
```diff
@@ -165,7 +167,8 @@ private void handleTask(Task task, Map<String, Object> parameters, boolean isRet
         long activityId = activityService.addTaskStarted(task, parameters);
         Map<String, Object> metadata = prepareTaskMetadata(task.getId(), activityId, isRetry);
 
-        TaskContext taskContext = new TaskContext(task, parameters, metadata, activityService);
+        taskContext = new TaskContext(task, parameters, metadata, activityService);
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java]: remove changes in this class

File: modules/tasks/tasks/pom.xml [hunk 7]
```diff
@@ -139,7 +145,7 @@
                             org.motechproject.tasks.repository;version=${project.version},
                             org.motechproject.tasks.service;version=${project.version},
                             org.motechproject.tasks.service.osgi;version=${project.version},
-                            org.motechproject.tasks.util;version=${project.version}
+                            org.motechproject.tasks.service.util;version=${project.version}
```
Comment (kmadej) [inline modules/tasks/tasks/pom.xml]: what's the reason for this change ?

File: modules/tasks/tasks-test-bundle/src/main/java/org/motechproject/testmodule/service/impl/TasksTestServiceImpl.java [hunk 8]
```diff
@@ -4,22 +4,30 @@
...
+
+        taskTestObjects.add(object);
         return object;
     }
 
     @Override
-    public TaskTestObject createTestObject(String name, String nameWithPrefix) {
-        return new TaskTestObject(name, nameWithPrefix);
+    public List<TaskTestObject> getTaskTestObjects() {
+
```
Comment (kmadej) [inline modules/tasks/tasks-test-bundle/src/main/java/org/motechproject/testmodule/service/impl/TasksTestServiceImpl.java]: remove empty line

File: modules/tasks/tasks/pom.xml (hunk 7 above)
Comment (pkornowski) [inline modules/tasks/tasks/pom.xml]: it is no longer needed change for my test, but there is no package org.motechproject.tasks.util . I should revert this change?

**Comment by kmadej** [comment]
test this please

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/TasksIntegrationTests.java [hunk 9]
```diff
@@ -4,7 +4,7 @@
 import org.junit.runners.Suite;
 
 @RunWith(Suite.class)
-@Suite.SuiteClasses({ChannelsDataServiceBundleIT.class,  TaskDataServiceBundleIT.class,
-                     TaskActivitiesDataServiceBundleIT.class, TasksBundleIT.class})
+@Suite.SuiteClasses({TasksBundleIT.class, ActionParametersBundleIT.class, ChannelsDataServiceBundleIT.class, TaskDataServiceBundleIT.class,
```
Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/TasksIntegrationTests.java]: why do you change order here ?

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersBundleIT.java [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+        List<TaskTestObject> fetchedTaskTestObjects = tasksTestService.getTaskTestObjects();
+
+        List<TaskTestObject> expectedTaskTestObject = prepareExpectedTaskTestObject();
+
+        assertTrue(fetchedTaskTestObjects.contains(expectedTaskTestObject.get(0)));
+        assertTrue(fetchedTaskTestObjects.contains(expectedTaskTestObject.get(1)));
+        assertTrue(fetchedTaskTestObjects.contains(expectedTaskTestObject.get(2)));
+    }
+
+    @After
```
Comment (kmadej) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/it/ActionParametersBundleIT.java]: this method should be below before method

File: modules/tasks/tasks/src/main/resources/META-INF/spring/blueprint.xml [hunk 11]
```diff
@@ -58,6 +58,7 @@
     <osgi:reference id="tasksDataServiceOSGi"
                     interface="org.motechproject.tasks.repository.TasksDataService"/>
 
+
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/META-INF/spring/blueprint.xml]: remove

File: modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java [hunk 12]
```diff
@@ -166,6 +166,7 @@ private void handleTask(Task task, Map<String, Object> parameters, boolean isRet
         Map<String, Object> metadata = prepareTaskMetadata(task.getId(), activityId, isRetry);
 
         TaskContext taskContext = new TaskContext(task, parameters, metadata, activityService);
+
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java]: remove

File: modules/tasks/tasks-test-bundle/src/main/resources/META-INF/spring/blueprint.xml [hunk 13]
```diff
@@ -10,6 +10,7 @@
     <context:annotation-config/>
     <context:component-scan base-package="org.motechproject.testmodule"/>
 
-    <osgi:service ref="tasksTestService"  interface="org.motechproject.testmodule.service.TasksTestService" auto-export="interfaces" />
+    
+    <osgi:service id="tasksTestServiceOsgi" ref="tasksTestService"  interface="org.motechproject.testmodule.service.TasksTestService" auto-export="interfaces" />
```
Comment (kmadej) [inline modules/tasks/tasks-test-bundle/src/main/resources/META-INF/spring/blueprint.xml]: is this id needed ?
